### ISSUE TITLE
Cannot checksum the latest version when creating CRAN packages

### ISSUE THREAD

**Issue by fanne-stat** [spack/spack#6136]
When creating the CRAN package fftwtools version 0.9-8, the most recent version can be checksumed is 0.9-7. (If the tar.gz is not from https://cran.r-project.org but from https://cran.rstudio.org, the latest version can be checksumed.)

```
spack create https://cran.r-project.org/src/contrib/fftwtools_0.9-8.tar.gz
  ==> This looks like a URL for fftwtools
  ==> Warning: Spack will not check SSL certificates. You need to update your Python to enable 
  certificate verification.
  ==> Found 6 versions of fftwtools:

  0.9-7  https://cran.r-project.org/src/contrib/Archive/fftwtools/fftwtools_0.9-7.tar.gz
  0.9-6  https://cran.r-project.org/src/contrib/Archive/fftwtools/fftwtools_0.9-6.tar.gz
  0.9-5  https://cran.r-project.org/src/contrib/Archive/fftwtools/fftwtools_0.9-5.tar.gz
  0.9-3  https://cran.r-project.org/src/contrib/Archive/fftwtools/fftwtools_0.9-3.tar.gz
  0.9-2  https://cran.r-project.org/src/contrib/Archive/fftwtools/fftwtools_0.9-2.tar.gz
  0.9-1  https://cran.r-project.org/src/contrib/Archive/fftwtools/fftwtools_0.9-1.tar.gz

  ==> How many would you like to checksum? (default is 1, q to abort)
```

**Comment by JusticeForMikeBrown** [comment]
It's asking if you want to get hash all of those versions.  If you do, meaning users can choose to install them, then choose 6. 

The resulting template:

```
    homepage = "http://www.example.com"
    url      = "https://cran.r-project.org/src/contrib/fftwtools_0.9-8.tar.gz"

    version('0.9-7', '202e0f9bfc9e1367dd2390a3c3d49159')
    version('0.9-6', 'e1f56069044c46a242b3873a41b14695')
    version('0.9-5', '3fc359f3cc4f4ac98a41f75c2319ab71')
    version('0.9-3', 'becf676f6e6cdb8cc73d5118e6d2bd9b')
    version('0.9-2', 'e38e13e7d61dde7d715d90b004b4e79b')
    version('0.9-1', '47a0c3f56b65a939688eb227e0b75f69')
```

**Comment by fanne-stat** [comment]
The problem is that the url used when creating package is for 0.9-8 but the versions to hash did not include 0.9-8.

**Comment by tgamblin** [comment]
I'm not sure why that happens -- seems like a bug in `spack create`.  But as a workaround, you can just run:

```
spack md5 https://cran.r-project.org/src/contrib/fftwtools_0.9-8.tar.gz
```

To get the missing hash.

**Comment by fanne-stat** [comment]
When you do `spack checksum` there is the same problem. The latest version does not appear in the list.

**Comment by adamjstewart** [comment]
Closed by #10758.
